ocs_ci/resiliency/network_faults.py

```diff
@@ -0,0 +1,317 @@
+                    log.warning(
+                        f"Could not verify qdisc status on {node.name}/{iface}: {e}"
+                    )
+
+        time.sleep(5)
+        log.info("All fault configurations attempted and verified.")
+
+    def network_packet_loss(self, percentage=25):
+        """Simulates packet loss on all nodes.
+
+        Args:
+            percentage (int): Packet loss percentage.
+        """
+        self._apply_fault(f"{percentage}% packet loss", f"loss {percentage}%")
+
+    def add_network_latency(self, delay_ms=100, jitter_ms=20):
+        """Simulates latency and jitter on all nodes.
+
+        Args:
+            delay_ms (int): Base delay in milliseconds.
+            jitter_ms (int): Jitter in milliseconds.
+        """
+        self._apply_fault(
+            f"{delay_ms}ms latency ±{jitter_ms}ms jitter",
+            f"delay {delay_ms}ms {jitter_ms}ms",
+        )
+
+    def add_packet_duplication(self, duplicate_percent=5):
+        """Simulates packet duplication on all nodes.
+
+        Args:
+            duplicate_percent (int): Packet duplication percentage.
+        """
+        self._apply_fault(
+            f"{duplicate_percent}% packet duplication",
+            f"duplicate {duplicate_percent}%",
+        )
+
+    def add_packet_corruption(self, corruption_percent=1):
+        """Simulates packet corruption on all nodes.
+
+        Args:
+            corruption_percent (int): Packet corruption percentage.
+        """
+        self._apply_fault(
+            f"{corruption_percent}% packet corruption", f"corrupt {corruption_percent}%"
+        )
+
+    def add_network_combine_faults(
+        self, delay_ms=80, jitter_ms=20, loss_percent=10, duplicate_percent=1
+    ):
+        """Applies a combination of network faults.
+
+        Args:
+            delay_ms (int): Base latency in milliseconds.
+            jitter_ms (int): Jitter in milliseconds.
+            loss_percent (int): Packet loss percentage.
+            duplicate_percent (int): Packet duplication percentage.
+        """
+        netem_cmd = (
+            f"delay {delay_ms}ms {jitter_ms}ms "
+            f"loss {loss_percent}% "
+            f"duplicate {duplicate_percent}%"
+        )
+        desc = f"combined faults (delay={delay_ms}ms±{jitter_ms}ms, loss={loss_percent}%, dup={duplicate_percent}%)"
+        self._apply_fault(desc, netem_cmd)
+
+    def pre_fault_injection_checks(self):
+        """Perform any pre-fault sanity checks such as node readiness."""
+        log.info("Performing pre-fault injection checks (placeholder)")
+
+    def post_fault_injection_checks(self):
+        """Verifies Ceph cluster health and recovers from node failures if needed."""
+        log.info("Verifying post-fault Ceph cluster health")
+        try:
+            if ceph_health_check(tries=3, delay=20):
+                log.info("Ceph cluster is healthy post-fault")
+                return
+        except Exception as e:
```

Comment:
Plz avoid using general exceptions. its highly recommended to create exceptions for specific cases like  CommandFailed or another expected conditions

Reply:
Done